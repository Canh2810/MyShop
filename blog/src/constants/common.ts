export const THEMES = {
  DARK: 'dark',
  LIGHT: 'light',
}

export const LOCAL_STORAGE_KEY = {
  AUTH: 'auth',
}

export const LIMIT_POSTS = '9'

export const QUERY_KEY = {
  POSTS: 'posts',
}
