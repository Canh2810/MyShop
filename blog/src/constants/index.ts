export * from './common'
export * from './menu'
export * from './mock-data'
export * from './routes'
export * from './messages'
export * from './regex'
