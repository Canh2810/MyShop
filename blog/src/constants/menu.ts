export const LIST_MENU_ITEM_MOCK = [
  { title: 'Home', href: '/' },
  { title: 'About', href: '/about' },
  { title: 'Contact', href: '/contact' },
]

export const HEADER_MENU = [
  { title: 'Home', href: '/' },
  { title: 'Blog', href: '/' },
  { title: 'Single Post', href: '/' },
  { title: 'Pages', href: '/' },
  { title: 'Contact', href: '/' },
]
