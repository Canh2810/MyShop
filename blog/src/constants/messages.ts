export const ERROR_MESSAGES = {
  FIELD_REQUIRED: (fieldName: string) => `${fieldName} is required`,
  FIELD_INVALID: (fieldName: string) =>
    `Invalid format of ${fieldName}, please check.`,
  LOGIN_INVALID: 'Invalid email or password',
  FAIL_TO_FETCH: 'Something went wrong!! Please try again.',
}
