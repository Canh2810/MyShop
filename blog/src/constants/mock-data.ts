export const POST_MOCK = {
  id: 'cvbvbdff3er3',
  category: 'Technology',
  title:
    'The Impact of Technology on the Workplace: How Technology is Changing',
  author: {
    avatar: 'https://www.w3schools.com/w3images/avatar2.png',
    username: 'Tracey Wilson',
  },
  date: 'August 20, 2022',
  imageURL:
    'https://images.unsplash.com/photo-1543877087-ebf71fde2be1?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8OXx8bmF0dXJhbHxlbnwwfHwwfHx8MA%3D%3D&auto=format&fit=crop&w=600&q=60',
}
