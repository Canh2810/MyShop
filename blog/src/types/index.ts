export * from './common'
export * from './posts'
export * from './forms'
export * from './users'
export * from './response'
