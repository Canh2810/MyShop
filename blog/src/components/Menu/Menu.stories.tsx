import { StoryObj, Meta } from '@storybook/react'
import Menu from '.'
import { MenuVariants } from '@/types'
import { LIST_MENU_ITEM_MOCK } from '@/constants'

export default {
  title: 'Components/Menu',
  component: Menu,
} as Meta

type Story = StoryObj<typeof Menu>

export const Default: Story = {
  args: {
    listItems: LIST_MENU_ITEM_MOCK,
  },
}

export const Vertical: Story = {
  args: {
    listItems: LIST_MENU_ITEM_MOCK,
    variant: MenuVariants.Vertical,
  },
}
