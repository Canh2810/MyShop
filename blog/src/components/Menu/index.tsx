// Libs
import { memo } from 'react'
import Link from 'next/link'

// Types
import { IMenuItem, MenuVariants } from '@/types'

export interface MenuProps {
  listItems: IMenuItem[]
  variant?: MenuVariants
  className?: string
}

const Menu = ({
  listItems,
  variant = MenuVariants.Horizontal,
  className = '',
}: MenuProps) => {
  const menuStyles = {
    [MenuVariants.Horizontal]: ' gap-[30px]',
    [MenuVariants.Vertical]: 'flex-col gap-2 ',
  }

  const menuItemStyles = {
    [MenuVariants.Horizontal]: 'font-primary  dark:text-white',
    [MenuVariants.Vertical]: 'font-tertiary  dark:text-light-400',
  }

  return (
    <ul
      className={`flex items-center justify-center ${menuStyles[variant]} ${className}`}
    >
      {listItems.map(({ title, href }: IMenuItem) => (
        <li
          key={title}
          className={`text-sm font-normal hover:opacity-70 text-dark-100 ${menuItemStyles[variant]}`}
        >
          <Link href={href}>{title}</Link>
        </li>
      ))}
    </ul>
  )
}

export default memo(Menu)
