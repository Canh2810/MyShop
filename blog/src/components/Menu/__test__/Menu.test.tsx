import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Menu from '..'
import { MenuVariants } from '@/types'
import { LIST_MENU_ITEM_MOCK } from '@/constants'

describe('Menu component', () => {
  it('should render correctly with variant is horizontal', () => {
    const { container } = render(<Menu listItems={LIST_MENU_ITEM_MOCK} />)
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with variant is vertical', () => {
    const { container } = render(
      <Menu listItems={LIST_MENU_ITEM_MOCK} variant={MenuVariants.Vertical} />,
    )
    expect(container).toMatchSnapshot()
  })

  it('calls the correct onClick handler when a menu item is clicked', () => {
    render(<Menu listItems={LIST_MENU_ITEM_MOCK} />)

    const menuItem = screen.getByText(LIST_MENU_ITEM_MOCK[0].title)

    userEvent.click(menuItem)
    expect(window.location.pathname).toBe(LIST_MENU_ITEM_MOCK[0].href)
  })
})
