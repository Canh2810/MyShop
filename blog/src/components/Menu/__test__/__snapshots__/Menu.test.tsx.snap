// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should render correctly with variant is horizontal 1`] = `
<div>
  <ul
    class="flex items-center justify-center  gap-[30px] "
  >
    <li
      class="text-sm font-normal hover:opacity-70 text-dark-100 font-primary  dark:text-white"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="text-sm font-normal hover:opacity-70 text-dark-100 font-primary  dark:text-white"
    >
      <a
        href="/about"
      >
        About
      </a>
    </li>
    <li
      class="text-sm font-normal hover:opacity-70 text-dark-100 font-primary  dark:text-white"
    >
      <a
        href="/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</div>
`;

exports[`Menu component should render correctly with variant is vertical 1`] = `
<div>
  <ul
    class="flex items-center justify-center flex-col gap-2  "
  >
    <li
      class="text-sm font-normal hover:opacity-70 text-dark-100 font-tertiary  dark:text-light-400"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="text-sm font-normal hover:opacity-70 text-dark-100 font-tertiary  dark:text-light-400"
    >
      <a
        href="/about"
      >
        About
      </a>
    </li>
    <li
      class="text-sm font-normal hover:opacity-70 text-dark-100 font-tertiary  dark:text-light-400"
    >
      <a
        href="/contact"
      >
        Contact
      </a>
    </li>
  </ul>
</div>
`;
